src/sections/image-with-text.liquid

```diff
@@ -0,0 +1,109 @@
+{%- comment -%}
+  This is a required section for the Shopify Theme Store.
+  It is available when you add the "Image with text" section in the theme editor.
+
+  Theme Store required settings
+  - Image
+  - Heading: title of the block
+  - Text: description of the block
+
+  Theme Store optional settings
+  - Button label: label associated with the button
+  - Button link: link of the button
+{%- endcomment -%}
+
+<section class="image-with-text">
+  {% capture image_layout %}
+    <div class="image-with-text__image">
+      {% if section.settings.image != blank %}
+        {% include 'responsive-image' with
+          image: section.settings.image,
+          max_width: 545,
+          max_height: 545
+        %}
+      {% else %}
+        {{ 'image' | placeholder_svg_tag: 'placeholder-svg' }}
+      {% endif %}
+    </div>
+  {% endcapture %}
+
+  {% if section.settings.layout == 'left' %}
+    {{ image_layout }}
+  {% endif %}
+
+  <div class="image-with-text__text">
+    {% if section.settings.title != blank %}
+      <h2>{{ section.settings.title | escape }}</h2>
+    {% endif %}
+    {% if section.settings.text != blank %}
+      <div>{{ section.settings.text }}</div>
+    {% endif %}
+    {% if section.settings.button_label != blank and section.settings.button_link != blank %}
+      <a href="{{ section.settings.button_link }}">
+        {{ section.settings.button_label | escape }}
+      </a>
+    {% endif %}
+  </div>
+
+  {% if section.settings.layout == 'right' %}
+    {{ image_layout }}
+  {% endif %}
+</section>
+
+{% schema %}
+  {
+    "name": "Image with text",
+    "class": "index-section",
+    "settings": [
+      {
+        "type": "image_picker",
+        "id": "image",
+        "label": "Image"
+      },
+      {
+        "type": "select",
+        "id": "layout",
+        "label": "Image alignment",
+        "default": "left",
+        "options": [
+          {
+            "value": "left",
+            "label": "Left"
+          },
+          {
+            "value": "right",
+            "label": "Right"
+          }
```

Comment:
Irrelevant to this PR but I don't feel like the values should be `left` and `right`, but rather `before` and `after` (or something similar) to describe the image position in relation to the text.

When setting the mobile version (in the theme editor) seeing `left` and `right` is confusing - the assumption would be `top` and `bottom` since the image and text is stacked.

Any suggestions @evija?

Reply:
I see your point here, however I think before/after is too vague. A question that comes up for me is "before what? where?". These questions aren't answered.
Also, we'd need to change the name of the setting - it's not really alignment anymore.. more hierarchy of content (I think). 
Before/after is vague enough that it works for both mobile and desktop, but I'm not we should be prioritizing this. We know desktop settings don't match the mobile version. But at least in this case you know exactly what to expect for desktop. 
You know the object (image), and you're choosing to align it to the left or right. 
If it's before/after, I'm choosing to align my object to before or after?